test/unit/test_metagenomics.py

```diff
@@ -29,6 +29,38 @@ def test_help_parser_for_each_command(self):
             parser = parser_fun(argparse.ArgumentParser())
             helpstring = parser.format_help()
 
+class TestKrakenUniqSummaryFilter(TestCaseWithTmp):
+
+    def _test_report(self, report_basename):
+        test_files_dir = util.file.get_test_input_path(self)
+        in_report = os.path.join(test_files_dir, 'input', report_basename)
+        out_report = util.file.mkstempfname('.txt')
+        
+        parser = metagenomics.parser_krakenuniq_report_filter(argparse.ArgumentParser())
+        args = parser.parse_args([in_report, out_report])
+        args.func_main(args)
+
+        # Check that results match (roughly) with what is expected
+        expected_out = os.path.join(test_files_dir, 'expected', report_basename)
+        self.assertApproxEqualValuesInDelimitedFiles(out_report, 
+                                                     expected_out, 
+                                                     dialect="tsv", 
+                                                     numeric_rel_tol=1e-3, 
+                                                     header_lines_to_skip=3)
+
+    def test_unchanged_report(self):
+        self._test_report("should_be_unchanged.txt")
+    
+    def test_leaves_trimmed(self):
+        self._test_report("should_have_leaves_trimmed.txt")
+
+    #def test_higher_node_trimmed_after_resumming(self):
+    #    self._test_report("should_have_higher_node_trimmed_after_resumming.txt")
```

Comment:
anything wrong with these tests at the moment?

Reply:
Nothing wrong--they're just not implemented yet (have to make the synthetic input/output).